astropy/utils/tests/test_data.py

```diff
@@ -430,62 +904,456 @@ def __init__(self, data):
+    assert is_url_in_cache(testurl)
+    assert not is_url_in_cache(testurl2)
+
+
+def test_export_import_roundtrip(tmpdir, temp_cache, valid_urls):
+    zip_file_name = os.path.join(tmpdir, "the.zip")
+    for u, c in itertools.islice(valid_urls, 7):
+        download_file(u, cache=True)
+    normal = check_download_cache()
+    initial_urls_in_cache = set(get_cached_urls())
+
+    export_download_cache(zip_file_name)
+    clear_download_cache()
+    import_download_cache(zip_file_name)
+
+    assert set(get_cached_urls()) == initial_urls_in_cache
+    assert check_download_cache(check_hashes=True) == normal
+
+
+def test_export_import_roundtrip_different_location(tmpdir, valid_urls):
+    original_cache = tmpdir / "original"
+    os.mkdir(original_cache)
+    zip_file_name = tmpdir / "the.zip"
+
+    urls = list(itertools.islice(valid_urls, 7))
+    initial_urls_in_cache = set(u for (u, c) in urls)
+    with paths.set_temp_cache(original_cache):
+        for u, c in urls:
+            download_file(u, cache=True)
+        assert set(get_cached_urls()) == initial_urls_in_cache
+        export_download_cache(zip_file_name)
+
+    new_cache = tmpdir / "new"
+    os.mkdir(new_cache)
+    with paths.set_temp_cache(new_cache):
+        import_download_cache(zip_file_name)
+        check_download_cache(check_hashes=True)
+        assert set(get_cached_urls()) == initial_urls_in_cache
+        for (u, c) in urls:
+            assert get_file_contents(download_file(u, cache=True)) == c
+
+
+def test_cache_size_is_zero_when_empty(temp_cache):
+    assert not get_cached_urls()
+    assert cache_total_size() == 0
+
+
+def test_cache_size_changes_correctly_when_files_are_added_and_removed(
+    temp_cache, valid_urls
+):
+    u, c = next(valid_urls)
+    clear_download_cache(u)
+    s_i = cache_total_size()
+    download_file(u, cache=True)
+    assert cache_total_size() == s_i + len(c)
+    clear_download_cache(u)
+    assert cache_total_size() == s_i
+
+
+def test_cache_contents_agrees_with_get_urls(temp_cache, valid_urls):
+    r = []
+    for i in range(7):
+        a, a_c = next(valid_urls)
+        a_f = download_file(a, cache=True)
+        r.append((a, a_c, a_f))
+    assert set(cache_contents().keys()) == set(get_cached_urls())
+    for (u, c, h) in r:
+        assert cache_contents()[u] == h
+
+
+def test_free_space_checker_huge(tmpdir):
+    with pytest.raises(OSError):
+        check_free_space_in_dir(tmpdir, 1_000_000_000_000_000_000)
+
+
+def test_get_free_space_file_directory(tmpdir):
+    fn = tmpdir / "file"
+    with open(fn, "w"):
+        pass
+    with pytest.raises(OSError):
+        get_free_space_in_dir(fn)
+    assert get_free_space_in_dir(tmpdir) > 0
+
+
+def test_download_file_bogus_settings(invalid_urls, temp_cache):
+    u = next(invalid_urls)
+    with pytest.raises(ValueError):
+        download_file(u, sources=[])
+
+
+def test_download_file_local_directory(tmpdir):
+    with pytest.raises(urllib.request.URLError):
```

Comment:
Just to clarify, this fails because `tmpdir` exists but cannot be downloaded? Can you please add a comment for this test?